test/Base_Tests/src/Semantic/Multi_On_Any_Spec.enso

```diff
@@ -0,0 +1,41 @@
+from Standard.Base import all
+from Standard.Test import all
+
+type My_Atom
+    Value a
+
+    method_1 self = "My_Atom.method_1"
+
+type My_Refinement
+    Value x
+
+    method_1 self = "My_Refinement.method_1"
+    is_nothing self = "My_Refinement.is_nothing"
+
+My_Refinement.from (that : My_Atom) -> My_Refinement =
+    My_Refinement.Value that
+
+prepare_atom =
+    (My_Atom.Value 42) : My_Atom & My_Refinement
+
+add_specs suite_builder =
+    x = prepare_atom
+
+    suite_builder.group "Multi_On_Any" group_builder->
+        group_builder.specify "call `method_1` on `My_Atom`, not on `My_Refinement`" <|
+            x.method_1 . should_equal "My_Atom.method_1"
+
+        group_builder.specify "switch the order, then indeed `method_1` is called on `My_Refinement`" <|
+            ((x : My_Refinement & My_Atom).method_1) . should_equal "My_Refinement.method_1"
+
+        group_builder.specify "call the base `is_nothing` coming from `Any`" <|
+            x.is_nothing . should_be_false
+
+        group_builder.specify "hide the refinement" <|
+            ((x : My_Atom|Nothing).is_nothing) . should_be_false
```

Comment:
Can we also check that if I switch the ordering on `is_nothing` it will also delegate to `My_Refinement.is_nothing`?

We have such test for `method_1` but not `is_nothing` - which is the interesting part as it is defined on `Any`.

Reply:
- the tests are copied from description of #12185
- they are written by you
- if you want another test, feel free to provide it
    - either in the original issue description
    - or directly as a commit to this PR
- added a test in 0f6327c3f7

Reply:
Thanks for adding the test.

Note however that after you changed to reporting warning instead of returning a different value, the last test:
```
        group_builder.specify "hide the refinement" <|
             ((x : My_Atom|Nothing).is_nothing) . should_be_false
```
is no longer checking anything useful (both `Any.is_nothing` and `My_Refinement.is_nothing` return the same thing `False` - so this test will succeed either way).